### DISCUSSION TITLE
Connect Prompt ID to GENERATIONS when integrate Langfuse in LangChain.

### DISCUSSION THREAD

**Discussion by hdp173** [langfuse/langfuse#2162] [Support] [answered]
Hello.
I'm integrating langfuse into langchain ChatOpenAI model like this.
```
....

trace = langfuse.trace(name="trace1")
langchain_callback = trace.get_langchain_handler()
model = ChatOpenAI(callbacks=[langchain_callback], ...)

chain = prompt | model | parser

chain.ainvoke(...)
```
I'm fetching prompt from langfuse and I want to connect prompt id to GENERATION.
Can you help me how to do it?
Thank you.
Regards.

**Comment by hassiebp** [comment, marked as answer]
Linking Langfuse prompts to Langchain executions has been one of the most requested features recently, and we have just released support for it 🫡 🥳

For more details, please check out:

- [Docs on how to integrate, see tabs for Langchain in examples](https://langfuse.com/docs/prompts/get-started#link-with-langfuse-tracing-optional)
- [Python SDK release](https://github.com/langfuse/langfuse-python/releases/tag/v2.49.0)
- [JS SDK release](https://github.com/langfuse/langfuse-js/releases/tag/v3.25.0)
